src/main/java/com/sap/oss/phosphor/fosstars/model/value/ArtifactVersions.java

```diff
@@ -0,0 +1,181 @@
+    return new ArtifactVersions();
+  }
+
+  /**
+   * Initializes a set of versions.
+   *
+   * @param versions A set of versions.
+   */
+  @JsonCreator
+  public ArtifactVersions(@JsonProperty("elements") Set<ArtifactVersion> versions) {
+    Objects.requireNonNull(versions, "versions can't be null!");
+    this.elements = new HashSet<>(versions);
+  }
+
+  /**
+   * Initializes a set of versions.
+   *
+   * @param versions A number of versions.
+   */
+  public ArtifactVersions(ArtifactVersion... versions) {
+    this(setOf(versions));
+  }
+
+  /**
+   * Get a size of the set.
+   *
+   * @return A number of versions in the set.
+   */
+  public int size() {
+    return elements.size();
+  }
+
+  /**
+   * Check if artifact versions set is empty.
+   *
+   * @return true if empty otherwise false.
+   */
+  public boolean empty() {
+    return elements.isEmpty();
+  }
+
+  /**
+   * Checks if the collection contains one of the other versions.
+   *
+   * @param versions The other versions.
+   * @return True if at least one of the other versions is present in the collection,
+   *          false otherwise.
+   */
+  public boolean containsAnyOf(ArtifactVersions versions) {
+    for (ArtifactVersion version : versions) {
+      if (this.elements.contains(version)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Get versions sorted by date.
+   * First entry is the latest release.
+   *
+   * @return versions sorted by date
+   */
+  public Collection<ArtifactVersion> getSortByReleaseDate() {
+    SortedSet<ArtifactVersion> sortedArtifacts = new TreeSet<>(RELEASE_DATE_COMPARISON);
+    sortedArtifacts.addAll(elements);
+    return sortedArtifacts;
+  }
+
+  /**
+   * Returns Artifact version with given version.
+   * If the version is not available an empty optional is returned.
+   *
+   * @param version to be searched version
+   * @return found version or empty optional
+   */
+  public Optional<ArtifactVersion> getArtifactVersion(String version) {
+    return elements.parallelStream()
+        .filter(v -> v.getVersion().equals(version))
+        .findFirst();
+  }
+
+  /**
+   * Get a set of versions.
+   *
+   * @return The versions.
+   */
+  @JsonGetter("elements")
+  public Set<ArtifactVersion> get() {
+    return new HashSet<>(elements);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o instanceof ArtifactVersions == false) {
+      return false;
+    }
+    ArtifactVersions other = (ArtifactVersions) o;
+    return Objects.equals(elements, other.elements);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(elements);
+  }
+
+  @Override
+  public String toString() {
+    Collection<ArtifactVersion> sortByReleaseDate = getSortByReleaseDate();
+    final int limit = 5;
+    String message = sortByReleaseDate.stream()
+        .limit(limit)
+        .map(v -> v.getVersion() + ":" + v.getReleaseDate())
```

Comment:
Not a big deal, up to you.

```suggestion
        .map(v -> String.format("%s:%s", v.getVersion(), v.getReleaseDate()))
```